app/models/voucher.rb

```diff
@@ -12,7 +14,8 @@ class Voucher < ApplicationRecord
 
   before_validation :strip_code
 
-  validates :code, length: { maximum: STRING_COLUMN_LIMIT },
+  validates :code, format: { without: INVALID_CODE_REGEX },
```

Comment:
I think that this is too much validation. The browser doesn't allow this, right? Is there another way this could happen by accident? I don't think that there's a case of some trying this intentionally and then it's their own fault. Going down this rabbit hole we could start to enforce only certain characters that are easy to print and read etc but I would leave this up to the person creating the voucher. It's always a balance between all the cool features and simple code.

Reply:
I agree; I appreciate the extra validation done here but in general validation is to help avoid mistakes or malicious behaviour. The browser input solves the problem for us, so we don't need to add additional code to the codebase. 
Especially as Maikel explained, we may need to handle newlines differently soon anyway.

Reply:
No problem, removed in https://github.com/openfoodfoundation/openfoodnetwork/pull/11259/commits/ab0d899a17c2198ee4d6de9e886816f3e75abd0e. This was in case vouchers could be created via an API endpoint in future.